botocore/docs/bcdoc/restdoc.py

```diff
@@ -205,7 +221,25 @@ def flush_structure(self):
                     self.style.link_target_definition(refname, link)
         value = self.getvalue()
         for name, section in self._structure.items():
-            value += section.flush_structure()
+            # Checks is the AWS API Documentation link has been generated.
+            # If it has been generated, it gets passed as a the doc_link parameter.
+            if b'`AWS API Documentation <' in value:
+                start = value.find(b'`AWS API Documentation <')
```

Comment:
Another general comment, I'm wondering if doing something with a [regex substitution](https://docs.python.org/3/library/re.html#re.sub) may be safer here?

Reply:
+1

Reply:
I updated this to use `re.search(<pattern>, <text>)`. See new changes for details.